include/revng/TypeNames/PTMLCTypeBuilder.h

```diff
@@ -542,6 +544,134 @@ class CTypeBuilder : public CBuilder {
                                  Actions);
   }
 
+  struct TagPair {
+    std::string Definition;
+    std::string Reference;
+  };
+
+  /// Special case handling for local variables.
+  ///
+  /// \note that both tags are acquired at once, this removes the need to keep
+  ///       track of already emitted automated names.
+  TagPair
+  getVariableTags(VariableNameBuilder &SubNameBuilder,
```

Comment:
1. Why is it a *sub* name builder? Let's keep things simple, if we can.
2. Please do not use the term tag (`TagPair`, `getVariableTags`).
   Refrain from using that term because it conveys zero information about the content.
   If I understand correctly what this method does, I think it'd be better to split it in `getVariableDefinition` and `getVariableUse`.
   If you really really want to keep these two together, use something like `getVariableDefinitionAndUseTokens`.
   The outputs are *tokens*, not *tags*.

Reply:
1. To stress that it's not the normal `NameBuilder` this access, but a counting one spawned from it.

2. Wait, this in fact produces a _tag_, _ptml tag_ to be exact.

I'm fine with `tag` -> `token` but if we change it, we should do so everywhere, not just here!

Reply:
1. To me the term `Sub` is more confusing than helpful. It implies how `VariableNameBuilder` is done internally and how is put together in the caller, which is not relevant here. I'd drop it, but maybe you have a better understanding and it's instead relevant.
2. Let's postpone this discussion.

Reply:
Done.